openeo/rest/userfile.py

```diff
@@ -0,0 +1,58 @@
+import typing
+from typing import Any, Dict, List, Union
+from pathlib import Path
+
+if typing.TYPE_CHECKING:
+    # Imports for type checking only (circular import issue at runtime).
+    from openeo.rest.connection import Connection
```

Comment:
This if-import is not necessary: you're not using `Connection` as a class 
(only as a type hint, but as a string, which works without import)

Reply:
Copied from Service, so we should check whether it can also be removed there.

Reply:
aha interesting
removed in 20e4c5e8a260a7ac45808ed0a145680245e226d4

Reply:
I stand corrected, linting with flake8 apparently requires this import even for string type hints.
https://github.com/Open-EO/openeo-python-client/actions/runs/4207089915/jobs/7301432268

Reverted in 32c70d03664eb74610e6f56067acc817556244c8